### ISSUE TITLE
Uncaught TypeError: Cannot read property 'start' of undefined

### ISSUE THREAD

**Issue by jacargentina** [remarkjs/react-markdown#250]
Using `with-html` i'm  getting this stack. Any advice?

```
html-parser.js:171 Uncaught TypeError: Cannot read property 'start' of undefined
    at parsedHtml (html-parser.js:171)
    at html-parser.js:78
    at overload (index.js:26)
    at one (index.js:27)
    at all (index.js:51)
    at one (index.js:35)
    at visitParents (index.js:21)
    at visit (index.js:21)
    at parseHtml (html-parser.js:38)
    at react-markdown.js:70
    at Array.reduce (<anonymous>)
    at ReactMarkdown (react-markdown.js:69)
    at ReactMarkdownWithHtml (with-html.js:13)
    at mountIndeterminateComponent (react-dom.development.js:15167)
    at beginWork (react-dom.development.js:15692)
    at performUnitOfWork (react-dom.development.js:18750)
    at workLoop (react-dom.development.js:18791)
    at HTMLUnknownElement.callCallback (react-dom.development.js:147)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:196)
    at invokeGuardedCallback (react-dom.development.js:250)
    at replayUnitOfWork (react-dom.development.js:17998)
    at renderRoot (react-dom.development.js:18909)
    at performWorkOnRoot (react-dom.development.js:19812)
    at performWork (react-dom.development.js:19722)
    at performSyncWork (react-dom.development.js:19696)
    at requestWork (react-dom.development.js:19551)
    at scheduleWork (react-dom.development.js:19358)
    at Object.enqueueForceUpdate (react-dom.development.js:12826)
    at Query.push../node_modules/react/cjs/react.development.js.Component.forceUpdate (react.development.js:373)
    at Query._this.updateCurrentData (react-apollo.browser.umd.js:461)
    at Object.next (react-apollo.browser.umd.js:440)
    at notifySubscription (Observable.js:152)
    at onNotify (Observable.js:196)
    at SubscriptionObserver.next (Observable.js:248)
    at ObservableQuery.js:425
    at Array.forEach (<anonymous>)
    at Object.next (ObservableQuery.js:424)
    at QueryManager.js:611
    at QueryManager.js:986
    at Array.forEach (<anonymous>)
    at QueryManager.js:985
    at Map.forEach (<anonymous>)
    at QueryManager.broadcastQueries (QueryManager.js:981)
    at Object.next (QueryManager.js:1064)
    at notifySubscription (Observable.js:152)
    at onNotify (Observable.js:196)
    at SubscriptionObserver.next (Observable.js:248)
    at dedupLink.js:94
    at Array.forEach (<anonymous>)
    at Object.next (dedupLink.js:93)
```

**Comment by jacargentina** [comment]
Just try to put this html (image with data-uri) on the demo https://rexxars.github.io/react-markdown/

```html
<img width="16" height="16" alt="star" src="data:image/gif;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7" />
```

**Comment by braska** [comment]
Can anyone to say something about this issue? Anyone planning to fix this error? Maybe someone can give some tips how to fix this? It is critical for my project. I can fix it, but don't know from what I should start.

**Comment by braska** [comment]
Error occurred on this line: https://github.com/rexxars/react-markdown/blob/409a8c4fbf3a27936a9297382199586da4b31c87/src/plugins/html-parser.js#L167

I tried to debug this thing with breakpoints and looks like there should be `fromNode.position.start` instead of `fromNode.node.position.start`.

![image](https://user-images.githubusercontent.com/3268276/52076380-9ae16000-259f-11e9-991b-732bc7d3a1e9.png)

But I am not familiar with codebase of this project. @rexxars, can you help to resolve this issue?

**Comment by wallawe** [comment]
Having the same issue, really hard to debug. Any solutions?

**Comment by blueprint1985** [comment]
I managed to solve this by removing any empty lines from the HTML-code that I wanted to parse. For example
```
<h2 className="subpage">First header</h2>

<p className="subpage">First paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>

<h2 className="subpage">Second header</h2>

<p className="subpage">Second paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
```
was changed to
```
<h2 className="subpage">First header</h2>
<p className="subpage">First paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
<h2 className="subpage">Second header</h2>
<p className="subpage">Second paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
```
and then the error disappeared. Maybe this should be fixed in a future fix.

**Comment by wallawe** [comment]
As @blueprint1985 mentioned above, I found that my issue was caused by weird spaces after certain tags, e.g. `<h2 id="sources">List of Sources</h2> ` (notice the space at the very end).

This causes the parser to split that line into two separate nodes and return an array of elements in the parseNode function in 'plugins/html-parser.js'. 

The solution I came up with which is pretty hacky is to put the following on line 131:

```
  if (Array.isArray(el) && typeof el[1] === 'string' && el[1].trim() === "") {
    el = el[0]
  }
```

The whole function now looks like this
![image](https://user-images.githubusercontent.com/1158544/55408947-08255900-552f-11e9-8c1f-4bb8d51e82bc.png)
